pytest_mh/utils/selinux.py

```diff
@@ -0,0 +1,73 @@
+"""SELinux context."""
+
+from __future__ import annotations
+
+import re
+
+__all__ = [
+    "SELinuxContext",
+]
+
+
+class SELinuxContext(object):
+    """
+    Result of ``ls -Z``
+    """
+
+    def __init__(self, user: str, role: str, type: str, sensitivity: str, categories: str) -> None:
+        self.user: str = user
+        """
+        SELinux user.
+        """
+
+        self.role: str = role
+        """
+        SELinux role.
+        """
+
+        self.type: str = type
+        """
+        SELinux type.
+        """
+
+        self.sensitivity: str = sensitivity
+        """
+        SELinux sensitivity level.
+        """
+
+        self.categories: str = categories
+        """
+        SELinux categories.
+        """
+
+    def __str__(self) -> str:
+        return f"({self.user}:{self.role}:{self.type}:{self.sensitivity}:{self.categories})"
+
+    def __repr__(self) -> str:
+        return str(self)
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, SELinuxContext):
+            raise TypeError(f"Unexpected type: {type(other)}, expecting SELinuxContext")
+        return (
+            self.user == other.user
+            and self.role == other.role
+            and self.type == other.type
+            and self.sensitivity == other.sensitivity
+            and self.categories == other.categories
+        )
+
+    @classmethod
+    def FromOutput(cls, stdout: str) -> SELinuxContext:
+        match = re.match(r"([^ ]+)\s+", stdout)
+
+        if not match:
+            raise ValueError("Unexpected value: expecting space separated string")
+
+        selinux_context = match.group(1)
+        labels = selinux_context.split(":")
+
+        if len(labels) != 5:
```

Comment:
```
$ ls -Z sssd_ssh 
unconfined_u:object_r:user_home_t:s0 sssd_ssh
```

Are `categories` really always present?

Reply:
What will `ls -Z` return if selinux is completely disabled?

Reply:
`SELinuxContext` will be `None`. The test case should check for the SELinux feature to be enabled to run this test (either completely or partially).

Reply:
Do you have `ls -Z` output example in this case?

Reply:
```
       -Z, --context
              print any security context of each file
```
What about other types of "security context"?
I don't know if stuff like 'AppArmor' will show up here is well?
Or is this specifically about "selinux"?

Reply:
> The test case should check for the SELinux feature to be enabled

Or maybe `selinux_context()` could check this and return 'None' if disabled?

Reply:
> Or is this specifically about "selinux"?

This is just about SELinux context. I don't know how AppArmor works and from what I heard Suse is moving away from it to SELinux.

> Or maybe selinux_context() could check this and return 'None' if disabled?

There, there is no way of knowing whether SELinux is enabled or SELinux labeling failed. We need a way to distinguish those cases during test execution. In this particular case the difference between these two cases is important because the shadow CI runs on Fedora, Debian, Alpine and OpenSuse. Some of these distributions do not provide SELinux.

Reply:
I don't know if I inadvertently resolved this conversation, so I'm reopening it. Feel free to resolve it if I clarified all your doubts.

Reply:
I don't see a "resolve" button, likely lack of access rights.